editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -6201,8 +6236,22 @@ void Node3DEditor::_finish_grid() {
 }
 
 void Node3DEditor::update_grid() {
-	_finish_grid();
-	_init_grid();
+	const Camera3D::Projection current_projection = viewports[0]->camera->get_projection();
+
+	if (current_projection != grid_camera_last_update_perspective) {
+		grid_init_draw = false; // redraw
+		grid_camera_last_update_perspective = current_projection;
+	}
+
+	// Gets a orthogonal or perspective position correctly (for the grid comparison)
+	const Vector3 camera_position =	get_editor_viewport(0)->camera->get_position();
```

Comment:
Style issue here, you're using a tab instead of a space after `=`.

Reply:
fixed (i just had not submitted change from git hook)